src/core/runtime.d

```diff
@@ -492,32 +562,48 @@ extern (C) bool runModuleUnitTests()
         }
     }
 
-    if( Runtime.sm_moduleUnitTester is null )
+    if (Runtime.sm_moduleUnitTester !is null)
+        return Runtime.sm_moduleUnitTester();
+    else if (Runtime.sm_boolModuleUnitTester !is null)
+        return Runtime.sm_boolModuleUnitTester() ? UnittestResult.pass :
+            UnittestResult.fail;
+    int failed = 0;
+    int nRun = 0;
+    foreach( m; ModuleInfo )
     {
-        size_t failed = 0;
-        foreach( m; ModuleInfo )
+        if( m )
         {
-            if( m )
-            {
-                auto fp = m.unitTest;
+            auto fp = m.unitTest;
 
-                if( fp )
+            if( fp )
+            {
+                ++nRun;
+                try
                 {
-                    try
-                    {
-                        fp();
-                    }
-                    catch( Throwable e )
-                    {
-                        _d_print_throwable(e);
-                        failed++;
-                    }
+                    fp();
+                }
+                catch( Throwable e )
+                {
+                    _d_print_throwable(e);
+                    ++failed;
                 }
             }
         }
-        return failed == 0;
     }
-    return Runtime.sm_moduleUnitTester();
+    import rt.config : rt_configOption;
+    if (failed != 0)
+        return -failed;
+    else switch (rt_configOption("unittestmode"))
+    {
+    case "":
+        // By default, run main. Switch to only doing unit tests in 2.074
+    case "runmain":
```

Comment:
Yes. I accidentally commented on the wrong line. I left the comment because I hate when I get a notification and then I get find it on GitHub because it was removed.